proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1790 @@
+
+  * Let `include-space` be true if `space` is null, and false otherwise.
+
+  * If `input` is a bracketed list, or a list with a separator other than
+    'slash' or 'space', throw an error.
+
+  * If `input` is a slash-separated list:
+
+    * If `input` doesn't have exactly two elements, throw an error.
+
+    * Otherwise, let `components` be the first element and `alpha` the second
+      element of `input`.
+
+  * Otherwise:
+
+    * Let `components` be an unbracketed space separated list of all except the
+      last element of `input`.
+
+    * If the last element of `input` is an unquoted string that contains `/`:
+
+      * Let `split-last` be the result calling `string.split()` with the last
+        element of `input` as the string to split, and `/` as the separator.
+
+      * If there are not two items in `split-last`, throw an error.
+
+      * If either item in `split-last` can be coerced to a number, replace the
+        current value of the item with the resulting number value.
+
+      * Let `alpha` be the second element in `split-last`, and append the first
+        element of `split-last` to `components`.
+
+      > This solves for a legacy handling of `/` in Sass that would produce an
+      > unquoted string when the alpha value is a CSS function such as `var()`
+      > or when either value is the keyword `none`.
+
+    * Otherwise, if the last element of `input` has preserved its status as two
+      slash-separated numbers:
+
+      * Let `alpha` be the number after the slash, and append the number before
+        the slash to `components`.
+
+    * Otherwise, append the last element of `input` to `components`.
+
+  * If `components` is an empty list, throw an error.
+
+  * If `components` is a [special variable string]:
+
+    * Let `channels` be the value of `components`.
+
+  * Otherwise:
+
+    * If `components` is not an unbracketed space-separated list, throw an error.
+
+    * If `space` is null:
+
+      * Let `input-space` be the first element in `components`.
+
+      * If `input-space` is not either a [known color space] or an unquoted
+        string, throw an error.
+
+      * Let `space` be the value of `input-space`.
+
+      * Let `channels` be an unbracketed space-separated list with the
+        remaining elements from `components`.
+
+    * Otherwise, let `channels` be the value of `components`.
+
+    * Let `expected` be the number of channels in `space` if `space` is a
+      [known color space], and null otherwise.
+
+    * If any element of channels is not either a number, a special variable
+      string, a special number string, or the keyword `none`, throw an error.
+
+  * If `alpha` is null, let `alpha` be `1`.
+
+  * Otherwise, If `alpha` is not a [special number string]:
+
+    * If `alpha` is a number, set `alpha` to the result of
+      [percent-converting] `alpha` with a max of 1, and then clamping the value
+      between 0 and 1, inclusive.
```

Comment:
Currently, "percent-converting a number" requires that the input be a percent—but at least in this case, we also want to support raw numbers that don't need percent-conversion.

Reply:
I'm pretty sure my definition of the percent-converting procedure accepts either - and only performs conversion on percentages.